### PR TITLE
Reverse resolving and refactor

### PR DESCRIPTION
**Pull request by balintligeti**

### CODE REVIEW

File: src/components/[guild]/EditGuild/components/Admins/Admins.tsx [hunk 1]
```diff
@@ -66,7 +66,7 @@ const Admins = () => {
     const ownerOption = {
       ...(options.find((o) => o.value === ownerAddress) ?? {
         value: ownerAddress,
-        label: shortenHex(ownerAddress),
+        label: ownerAddress.length === 42 ? shortenHex(ownerAddress) : ownerAddress,
```
Comment (BrickheadJohnny) [inline src/components/[guild]/EditGuild/components/Admins/Admins.tsx:69]: There's an `ADDRESS_REGEX` regex which we use to validate addresses in our application, you could use that here too

File: src/components/[guild]/EditGuild/components/Admins/Admins.tsx [hunk 2]
```diff
@@ -81,7 +81,10 @@ const Admins = () => {
           return {
             ...(option ?? {
               value: admin,
-              label: shortenHex(admin),
+              label:
+                admin.startsWith("0x") && admin.length === 42
```
Comment (BrickheadJohnny) [inline src/components/[guild]/EditGuild/components/Admins/Admins.tsx:85]: You could use the `ADDRESS_REGEX` here too

File: src/components/[guild]/EditGuild/components/Admins/components/AdminSelect.tsx [hunk 3]
```diff
@@ -8,20 +8,63 @@ import {
...
+import { Bug } from "phosphor-react"
+import { PropsWithChildren, useEffect } from "react"
+import { useFormContext, useWatch } from "react-hook-form"
 
 type PropsHelper = MultiValueGenericProps<unknown, boolean, GroupBase<unknown>>
+const CustomMultiValueContainer = ({
+  children,
+  ...multiValueContainerProps
+}: PropsWithChildren<PropsHelper>) => {
+  const domain = multiValueContainerProps.data.value.startsWith("0x")
```
Comment (BrickheadJohnny) [inline src/components/[guild]/EditGuild/components/Admins/components/AdminSelect.tsx:21]: Also a good place to use `ADDRESS_REGEX`

File: src/components/common/CopyableAddress.tsx [hunk 4]
```diff
@@ -19,7 +25,7 @@ const CopyableAddress = ({ address, decimals = 3, ...rest }: Props): JSX.Element
       hasArrow
     >
       <Button onClick={onCopy} variant="unstyled" height="auto" {...rest}>
-        <Text>{shortenHex(address, decimals)}</Text>
+        <Text>{domain ? domain : shortenHex(address, decimals)}</Text>
```
Comment (BrickheadJohnny) [inline src/components/common/CopyableAddress.tsx:28]: Nit: `domain || shortenHex(address, decimals)`

File: src/hooks/resolving/useResolveAddress.ts [hunk 5]
```diff
@@ -0,0 +1,71 @@
...
+    const NNS = await fetchNNSName(provider, account) // "0xe5358cab95014e2306815743793f16c93a8a5c70"
+    if (NNS) return NNS
+
+    const ENS = await fetchENSName(provider, account) // "0x5df52E6B70F25919Ad29add390EFE2614f91b2C6"
+    if (ENS) return ENS
+
+    const lens = await fetchLensProtocolName(account) // "0xe055721b972d58f0bcf6370c357879fb3a37d2f3"
+    if (lens) return lens
+
+    const unstoppableResolver = new Resolution()
```
Comment (BrickheadJohnny) [inline src/hooks/resolving/useResolveAddress.ts:51]: I think you could move this into the `fetchUnstoppableName` function, so it works the same way as the other resolver functions

File: src/hooks/resolving/useResolveAddress.ts [hunk 6]
```diff
@@ -0,0 +1,71 @@
...
+
+    const unstoppableDomain = await fetchUnstoppableName(
+      unstoppableResolver,
+      account
+    ) // "0x94ef5300cbc0aa600a821ccbc561b057e456ab23"
+    if (unstoppableDomain) return unstoppableDomain
+
+    const dotbitResolver = createInstance()
```
Comment (BrickheadJohnny) [inline src/hooks/resolving/useResolveAddress.ts:59]: Same here

**Review by BrickheadJohnny — CHANGES_REQUESTED** [review]

**Review by BrickheadJohnny — APPROVED** [review]
Thanks for the updates, now it looks good to me. We'll only need to remove the unnecessary comments (test addresses) before merging it to main.

**Review by dovalid — APPROVED** [review]

**Pull request merged**